internal/mode/static/telemetry/collector_test.go

```diff
@@ -239,6 +253,18 @@ var _ = Describe("Collector", Ordered, func() {
 		})
 	})
 
+	Describe("clusterID collector", func() {
+		When("collecting clusterID", func() {
+			It("throws an error when collecting clusterID", func() {
+				expectedError := errors.New("there was an error getting clusterID")
+				k8sClientReader.GetReturns(expectedError)
```

Comment:
Hmm this refactor is interesting, I would love for there to be a better way to add GetCalls to the base. However, I have no idea how your code with `mergeGetCallsWithBase` works. Would you be able to specify where in the code it actually merges the GetCalls with the base, unsure if its a type but I can't determine where it does.

Reply:
@bjee19 
it calls the BaseCalls first.
Then it calls what you pass to the mergeGetCallsWithBase

This way, what you pass overrides the basecalls.

https://github.com/pleshakov/nginx-gateway-fabric/commit/3f1afc306975076b01933069c0826eee83dc6dbe#diff-d6fd8590c10477109d902536ba4665c4431bf44a563a192051cdf399133b8993R149-R152

Reply:
Ahh okay I'm understanding it more, I think this would be a good refactor. @salonichf5 you should probably do this first as both Kate and I's comments on changing the GetReturns to GetCalls gets affected by this

Reply:
I will try to do this for using the GetCall stubs.